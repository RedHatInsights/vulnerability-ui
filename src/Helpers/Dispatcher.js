import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import ReducerRegistry from '../Utilities/ReducerRegistry';
export function dispatchAction(actionCreator) {
    const store = ReducerRegistry.getStore();
    return store.dispatch(actionCreator).catch(res => {
        if ('json' in res) {
            let parsedMessage = res.json();
            parsedMessage.then(arr => {
                for (let err of arr.errors) {
                    store.dispatch(
                        addNotification({
                            variant: 'danger',
                            title: 'Error ' + err.status,
                            description: err.detail
                        })
                    );
                }
            });
        } else {
            store.dispatch(
                addNotification({
                    variant: 'danger',
                    title: 'Error ' + res.status,
                    description: res.statusText
                })
            );
        }
    });
}
