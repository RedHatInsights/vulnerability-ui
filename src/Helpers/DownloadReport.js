import { downloadFile } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';

class DownloadReport  {
    constructor() {
        this.defaultParams = {
            page: 1,
            page_size: Number.MAX_SAFE_INTEGER,
            date: new Date().toISOString().replace(/[T:]/g, '-').split('.')[0] + '-utc'
        };

        this.params = {
            cve: ''
        };

        this.pages = ['systems-exposed', 'system-cves', 'cves', 'system-list'];
        this.formats = ['csv', 'json'];
    }

    filename = (page) => {
        switch (page) {
            case 'systems-exposed':
                return `vulnerability_systems-exposed-${this.params.cve}--${this.defaultParams.date}`;
            case 'system-list':
                return `vulnerability_systems--${this.defaultParams.date}`;
            default:
                return `vulnerability_cves--${this.defaultParams.date}`;

        }
    }

    checkTypes = (fetchData, page, format) => {
        if (typeof fetchData !== 'function') {
            throw `exec expects the 1st param to be funtion. Received ${typeof fetchData}`;
        }

        if (!this.pages.includes(page)) {
            throw `Page ${page} does not exist`;
        }

        if (!this.formats.includes(format)) {
            throw `Format ${format} is not supported`;
        }
    }

    /**
     * Execute and download the report
     *
     * @param {Function} fetchData - Function to be called in order to fetch data.
     * @param {Object || Array} params - Object to be sent as request params. If you need to pass extra argument, pass an array.
     * @param {String} format - Supported formats to download are CSV and JSON.
     * @param {String} page - Name of the page based on which the filename will be determined.
     * @returns {void}
     */
    exec = async (fetchData, params, format, page) => {
        this.checkTypes(fetchData, page, format);
        this.params.cve = Array.isArray(params) ? params[0] : '';

        try {
            let { payload } = Array.isArray(params)
                ? await fetchData(this.params.cve, {
                    ...params[1],
                    ...this.defaultParams,
                    data_format: format
                }) :
                await fetchData({
                    ...params,
                    ...this.defaultParams,
                    data_format: format
                });

            let { data: response } = await payload;
            let data = format === 'json' ? JSON.stringify(response) : response;
            // downaloads the files
            downloadFile(data, this.filename(page), format);
        } catch (error) {
            throw `${error}`;
        }

    }

}

export default new DownloadReport();
