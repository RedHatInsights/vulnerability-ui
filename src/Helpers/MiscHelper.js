import { SecurityIcon, UnknownIcon } from '@patternfly/react-icons';
import { SortByDirection } from '@patternfly/react-table';
import findIndex from 'lodash/findIndex';
import propTypes from 'prop-types';
import React from 'react';
import { impactColorList, impactList, PUBLIC_DATE_OPTIONS } from './constants';
import qs from 'query-string';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

export const dataShape = propTypes.shape({
    data: propTypes.oneOfType([propTypes.object, propTypes.array]),
    meta: propTypes.object,
    isLoading: propTypes.bool
});

/**
 * Based on the  allowedParams it will construct the arguments for the API call
 * which is defined in the vulnerabilities-client. vulnerabilities-client is written is typescript
 * so we need to define and pass every single parameter. Order in allowedParams is important
 *
 * @param {Object} apiProps
 * @param {Array} allowedParams - order should align with the params in the vulnerabilities-client
 */
export function constructParameters(apiProps, allowedParams) {
    if (apiProps) {
        Object.keys(apiProps).forEach(
            key => (
                apiProps[key] === undefined
                || apiProps[key] === ''
            )
            && delete apiProps[key]
        );

        const params = allowedParams.map(item => apiProps[item]);

        return params;
    }

    return [];
}

// TODO DRY:similar to constructParameters
export function constructURLParameters(urlParams, allowedParams) {
    if (urlParams) {
        const params = { ...urlParams };
        Object.keys(urlParams).forEach(
            key => (
                params[key] === undefined
                || params[key] === ''
                || !allowedParams.includes(key)
                || params[key] === false
            )
            && delete params[key]

        );
        return params;
    }
}

export function formatDate(date = new Date(), includeTime = false) {
    const prepend = (number) => `${`${number}`.length === 1 ? '0' : ''}${number}`;
    const toFormat = new Date(date);
    if (toFormat instanceof Date && !isNaN(toFormat)) {
        const year = toFormat.getFullYear();
        const month = toFormat.getMonth() + 1; // month is zero indexed
        const day = toFormat.getDate();
        const hour = toFormat.getUTCHours();
        const minute = toFormat.getUTCMinutes();

        return includeTime
            ? `${year}-${prepend(month)}-${prepend(day)} ${prepend(hour)}:${prepend(minute)} UTC`
            : `${year}-${prepend(month)}-${prepend(day)}`;
    }

    return toFormat;
}

/**
 * Contructs the filters which have more than one parameter that needs to be passed
 * and resets all the parameteres when it's needed
 *
 * @param {Object} filterParams
 * @return {Object}
 */
export function constructFilterParameters(filterParams) {
    if (Object.prototype.hasOwnProperty.call(filterParams, 'publish_date')) {
        let publicEntry = PUBLIC_DATE_OPTIONS.find(item => item.value === filterParams.publish_date);

        if (typeof publicEntry !== 'undefined') {
            filterParams.public_from = publicEntry.from && formatDate(publicEntry.from);
            filterParams.public_to = publicEntry.to && formatDate(publicEntry.to);
        } else {
            filterParams.public_from = undefined;
            filterParams.public_to = undefined;
            filterParams.publish_date = undefined;
        }
    }

    return filterParams;
}

export function getImpactDetails(impact) {
    if (Object.keys(impactList).includes(impact)) {
        return {
            title: impactList[impact].title,
            color: impactList[impact].color,
            text: impactList[impact].text,
            titleContent: (
                <SecurityIcon size="lg" />
            )
        };
    } else {
        return {
            title: 'Unknown',
            color: impactColorList.unknown,
            text: '',
            titleContent: (
                <UnknownIcon size="lg" />
            )
        };
    }
}

export const createSortBy = (header, value) => {
    if (value) {
        let direction = value.indexOf('-') === 0 ? SortByDirection.desc : SortByDirection.asc;
        value = value.replace(/^(-|\+)/, '');
        const index = findIndex(header, item => item.key === value);
        let sort = {
            index,
            direction
        };
        return sort;
    }

    return {};
};

export const handleSortColumn = (key, direction, header, currentSort, apply) => {
    let columnName = header[key].key;
    const useDefault = currentSort && currentSort.substr(1) !== columnName;
    if (direction === SortByDirection.desc || useDefault) {
        columnName = '-' + columnName;
    }

    apply({ sort: columnName });
};

export const updateStateSet = (stateSet, names, value) => {
    [].concat(names).forEach(name => {
        value ? stateSet.add(name) : stateSet.delete(name);
    });
    return stateSet;
};

export const useUrlParams = (allowedParams) => {
    const url = new URL(window.location);
    const urlParams = qs.parse(url.search);

    const setUrlParams = (parameters) => {
        const searchParams = qs.stringify(constructURLParameters(parameters, allowedParams));
        window.history.replaceState(null, null, `${url.origin}${url.pathname}?${searchParams}`);
    };

    return [urlParams, setUrlParams];
};

export const updateRef = (meta, apply) => {
    const pages = parseInt(meta.pages);
    const page = parseInt(meta.page);
    const cvesCount = parseInt(meta.cvesCount);
    const goTo = (pages === page && page > 1 && cvesCount === 1) ? (pages - 1) : page;

    apply({ page: goTo });
};

export const mountWithIntl = (Component) => {
    const wrapper = mount(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const shallowWithIntl = (Component) => {
    const wrapper = shallow(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const insertIf = (condition, ...elements) => {
    return condition ? elements : [];
};
