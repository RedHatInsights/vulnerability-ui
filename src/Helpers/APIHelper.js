let BASE_ROUTE = '/api/vulnerability/v1';
import { GitApi } from '@redhat-cloud-services/vulnerabilities-client';
import { SystemProfileApi } from '@redhat-cloud-services/host-inventory-client';
import instance from '../Utilities/interceptors';
import { constructParameters } from './MiscHelper';

let api = new GitApi(undefined, BASE_ROUTE, instance);

const INVENTORY_API_BASE = '/api/inventory/v1';
const systemProfile = new SystemProfileApi(undefined, INVENTORY_API_BASE, instance);

const defaultParams = [
    'filter',
    'limit',
    'offset',
    'page',
    'page_size',
    'sort'
];

export function getCveDetails(synopsis) {
    const result = api.getCveDetails(synopsis);
    return result;
}

export function getSystemDetails(system) {
    const result = api.getSystemDetails(system);
    return result;
}

export function setSystemCveStatus(params) {
    let result = api.setStatus(params);
    return result;
}

export function optOutSystems(systems, optOut) {
    const data = { opt_out: optOut, inventory_id: new Array(...systems) };
    let result = api.setSystemsOptOut(data);
    return result;
}

export function setBusinessRisk(params) {
    let result = api.setCveBusinessRisk(params);
    return result;
}

export function setCveStatus(params) {
    let result = api.setCveStatus(params);
    return result;
}

/**
 * /vulnerabilities/cves
 * /vulnerabilities/cves/ids
 */
let cveParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'data_format',
    'business_risk_id',
    'status_id',
    'rule_presence',
    'tags',
    'sap_sids',
    'sap_system',
    'known_exploit',
    'affecting',
    'rhel_version'
];

export function getCveListByAccount(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'report',
        'advanced_report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveList(...parameterArray);
    return result;
}

export function getCveIdsList(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveIdsList(...parameterArray);
    return result;
}

/**
 * /cves/ID/affected_systems
 */
let systemsByCVEparams = [
    'status_id',
    'data_format',
    'uuid',
    'rule_key',
    'rule_presence',
    'rule',
    'tags',
    'sap_sids',
    'sap_system',
    'show_advisories',
    'advisory',
    'rhel_version',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getAffectedSystemsByCVE({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'report',
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsByCve(id, ...parameterArray);
    return result;
}

export function getAffectedSystemsIdsByCve({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsIdsByCve(id, ...parameterArray);
    return result;
}

/**
 * /systems/ID/cves
 * /systems/ID/cves/ids
 */
let cveBySystemParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'status_id',
    'data_format',
    'business_risk_id',
    'rule_presence',
    'show_advisories',
    'advisory',
    'rule_key',
    'known_exploit',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getCveListBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams,
        'report'
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveListBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

export function getCveIdsBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveIdsBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

/**
 * /systems/ids
 * /systems
 */
let systemsParams = [
    'data_format',
    'stale',
    'uuid',
    'tags',
    'sap_sids',
    'sap_system',
    'excluded',
    'rhel_version'
];

export function getSystems(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams,
        'report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsList(...parameterArray);
    return result;
}

export function getSystemsIds(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsIds(...parameterArray);
    return result;
}

export function getAnnouncement() {
    let result = api.getAnnouncement();
    return result;
}

export function getDashbar(apiProps) {
    let parameterNames = [
        'tags',
        'sap_sids',
        'sap_system',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getDashbar(...parameterArray);
    return result;
}

export function getOperatingSystems() {
    return systemProfile.apiSystemProfileGetOperatingSystem();
}

export function fetchPdfReport(template, filters) {
    const CRC_PDF_GENERATE_API = '/api/crc-pdf-generator/v1/generate';
    const url = new URL(CRC_PDF_GENERATE_API,  window.location.origin);

    return fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },

        body: JSON.stringify({
            service: 'vulnerability',
            template,
            params: {
                ...filters
            }
        })
    }, 50000)
        .then((response) => response.blob());
}
