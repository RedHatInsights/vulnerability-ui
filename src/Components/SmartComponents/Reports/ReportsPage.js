import React,  { useState } from 'react';
import { Grid, GridItem, Card, CardTitle, CardBody, CardFooter } from '@patternfly/react-core';
import { Main } from '@redhat-cloud-services/frontend-components';
import { FileAltIcon } from '@patternfly/react-icons';
import DownloadExecutive from './DownloadExecutive';
import ReportConfigModal from '../Modals/ReportConfigModal';
import messages from '../../../Messages';
import { intl } from '../../../Utilities/IntlProvider';
import { ChartPieSolid } from '../../PresentationalComponents/CustomIcons/CustomIcons';
import Header from '../../PresentationalComponents/Header/Header';
import DownloadCVEsReport from '../Reports/DownloadCVEsReport';
import { constructFilterParameters, buildFilters } from '../Reports/ReportsHelper';
import { CVE_REPORT_FILTERS, DEFAULT_FILTER_DATA } from '../../../Helpers/constants';
import styles from './Common/styles';

const ReportsPage = () => {
    console.log('testing');
    const [isModalOpen, setModalOpen] = useState(false);
    const [downloadCVEReport, setDownloadCVEReport] = useState(false);
    const [reportTitle, setReportTitle] = useState(intl.formatMessage(messages.customReportDefaultTitle));
    const [userNotes, setUserNotes] = useState('');
    const [columnsToInclude, setColumnsToInclude] = useState(Object.keys(CVE_REPORT_FILTERS));
    const [filterData, setFilterData] = useState(DEFAULT_FILTER_DATA);

    const handleModalClose = () => {
        setReportTitle(intl.formatMessage(messages.customReportDefaultTitle));
        setUserNotes('');
        setColumnsToInclude(Object.keys(CVE_REPORT_FILTERS));
        setFilterData(DEFAULT_FILTER_DATA);
        setDownloadCVEReport(false);
        setModalOpen(false);
    };

    const handleDownloadButton = () => {
        setDownloadCVEReport(true);
        setModalOpen(false);
    };

    return (
        <React.Fragment>
            <Header title={intl.formatMessage(messages.reportsPageTitle)} showBreadcrumb={false}/>
            <Main>
                <Grid hasGutter lg={3} md={4} sm={12}>
                    <GridItem>
                        <Card className="report-card">
                            <CardTitle>
                                <ChartPieSolid style={styles.pieChartIcon}/>
                                <span className="pf-u-ml-sm" style={styles.cardTitle}>
                                    {intl.formatMessage(messages.executiveReportCardTitle)}
                                </span>
                            </CardTitle>
                            <CardBody>
                                {intl.formatMessage(messages.executiveReportCardDescription)}
                            </CardBody>
                            <CardFooter>
                                <DownloadExecutive/>
                            </CardFooter>
                        </Card>
                    </GridItem>
                    <GridItem>
                        <Card className="report-card">
                            <CardTitle>
                                <FileAltIcon size="lg" color="var(--pf-global--link--Color)"/>
                                <span className="pf-u-ml-sm" style={styles.cardTitle}>
                                    {intl.formatMessage(messages.customReportCardTitle)}
                                </span>
                            </CardTitle>
                            <CardBody>
                                {intl.formatMessage(messages.customReportCardDescription)}
                            </CardBody>
                            <CardFooter>
                                <a className="create-report"
                                    onClick={() => setModalOpen(true)}>{intl.formatMessage(messages.customReportCardButton)}
                                </a>
                            </CardFooter>
                        </Card>
                    </GridItem>
                </Grid>
            </Main>
            <ReportConfigModal
                isOpen={isModalOpen}
                columnsToInclude={columnsToInclude}
                setColumnsToInclude={setColumnsToInclude}
                filterData={filterData}
                userNotes={userNotes}
                reportTitle={reportTitle}
                setReportTitle={setReportTitle}
                setUserNotes={setUserNotes}
                setFilterData={setFilterData}
                handleModalClose={(handleModalClose)}
                handleDownloadButton={handleDownloadButton}
            />

            { downloadCVEReport && <DownloadCVEsReport
                type='cves'
                key="download"
                reportData={{
                    header: columnsToInclude,
                    reportTitle,
                    userNotes
                }}
                showButton={false}
                onSuccess={handleModalClose}
                params={constructFilterParameters(filterData)}
                filters={buildFilters(filterData)}
                isReportDynamic
                label={messages.configModalExportReport}
            />}
        </React.Fragment>
    );
};

export default ReportsPage;
