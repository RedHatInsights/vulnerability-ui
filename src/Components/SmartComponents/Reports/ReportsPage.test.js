import toJson from 'enzyme-to-json';
import { mountWithIntl } from '../../../Helpers/MiscHelper';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import ReportsPage from './ReportsPage';

/* Mock function returning permissions */
jest.mock('../../../Helpers/Hooks', () => ({
    ...jest.requireActual('../../../Helpers/Hooks'),
    useRbac: () => [[true, true, true], false]
}));

jest.useFakeTimers('modern').setSystemTime(new Date('2020-01-01').getTime());

const customMiddleWare = store => next => action => {
    next(action);
}

const mockStore = configureStore([customMiddleWare]);
const store = mockStore({});
window.insights = { loadRemediations: jest.fn(), chrome: { getUserPermissions: jest.fn() } };

let wrapper;
beforeEach(() => {
    wrapper = mountWithIntl(
        <Provider store={store}>
            <Router>
                <ReportsPage />
            </Router>
        </Provider>
    );
});

describe('Reports page component', () => {
    it('Should match snapshots', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should open configuration modal upon clicking button in the second card and close upon clicking cancel.', () => {
        wrapper.find('.create-report').simulate('click');
        expect(wrapper.find('ReportConfigModal').props().isOpen).toBeTruthy();

        wrapper.find('ReportConfigModal').find('button').last().simulate('click')
        expect(wrapper.find('ReportConfigModal').props().isOpen).toBeFalsy();
    });

    it('Should open configuration modal upon clicking button in the second card and close and start download upon clicking export.', () => {
        expect(wrapper.find('DownloadCVEsReport')).toHaveLength(0);

        wrapper.find('.create-report').simulate('click');
        expect(wrapper.find('ReportConfigModal').props().isOpen).toBeTruthy();

        const buttons = wrapper.find('ReportConfigModal').find('button');
        buttons.at(buttons.length - 1).simulate('click');

        expect(wrapper.find('ReportConfigModal').props().isOpen).toBeFalsy();
    });
});