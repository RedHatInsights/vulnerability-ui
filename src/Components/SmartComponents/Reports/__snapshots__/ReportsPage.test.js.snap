// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports page component Should match snapshots 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ReportsPage>
        <Page
          defaultManagedSidebarIsOpen={true}
          isManagedSidebar={false}
          mainTabIndex={-1}
          onPageResize={[Function]}
        >
          <div
            className="pf-c-page"
          >
            <main
              className="pf-c-page__main"
              tabIndex={-1}
            >
              <withRouter(Connect(VulnerabilityHeader))
                showBreadcrumb={false}
                title="Reports"
              >
                <Connect(VulnerabilityHeader)
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  showBreadcrumb={false}
                  title="Reports"
                >
                  <VulnerabilityHeader
                    dispatch={[Function]}
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    showBreadcrumb={false}
                    title="Reports"
                  >
                    <PageHeader>
                      <section
                        className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
                        widget-type="InsightsPageHeader"
                      >
                        <PageHeaderTitle
                          title="Reports"
                        >
                          <Component
                            className=""
                            headingLevel="h1"
                            size="2xl"
                            widget-type="InsightsPageHeaderTitle"
                          >
                            <h1
                              className="pf-c-title pf-m-2xl"
                              widget-type="InsightsPageHeaderTitle"
                            >
                               
                              Reports
                               
                            </h1>
                          </Component>
                        </PageHeaderTitle>
                      </section>
                    </PageHeader>
                  </VulnerabilityHeader>
                </Connect(VulnerabilityHeader)>
              </withRouter(Connect(VulnerabilityHeader))>
              <Connect(Main)>
                <Main>
                  <section
                    className="pf-l-page__main-section pf-c-page__main-section"
                    page-type=""
                  >
                    <Component
                      hasGutter={true}
                    >
                      <div
                        className="pf-l-grid pf-m-gutter"
                      >
                        <Component
                          span={3}
                        >
                          <div
                            className="pf-l-grid__item pf-m-3-col"
                          >
                            <Component>
                              <article
                                className="pf-c-card"
                              >
                                <Component>
                                  <div
                                    className="pf-c-card__title"
                                  >
                                    <ChartPieSolid>
                                      <img
                                        alt="Static pie solid"
                                        src="test-file-stub"
                                      />
                                    </ChartPieSolid>
                                    <span
                                      className="pf-u-ml-sm"
                                      style={
                                        Object {
                                          "verticalAlign": "0.3rem",
                                        }
                                      }
                                    >
                                      Executive report
                                    </span>
                                  </div>
                                </Component>
                                <Component>
                                  <div
                                    className="pf-c-card__body"
                                  >
                                    An executive summary of vulnerabilities identified by Red Hat that may impact your RHEL servers.
                                  </div>
                                </Component>
                                <Component>
                                  <div
                                    className="pf-c-card__footer"
                                  >
                                    <DownloadExecutive>
                                      <DownloadButton
                                        allPagesHaveTitle={false}
                                        asyncFunction={[Function]}
                                        buttonProps={
                                          Object {
                                            "isInline": true,
                                            "variant": "link",
                                          }
                                        }
                                        fileName="vulnerability_executive-report--2020-01-01.pdf"
                                        isPreview={false}
                                        label="Download PDF"
                                        pages={Array []}
                                        reportName="Executive report:"
                                        size="A4"
                                        title=""
                                        type="Vulnerability"
                                      >
                                        <Component
                                          isInline={true}
                                          onClick={[Function]}
                                          variant="link"
                                        >
                                          <button
                                            aria-disabled={null}
                                            aria-label={null}
                                            className="pf-c-button pf-m-link pf-m-inline"
                                            data-ouia-component-id={null}
                                            data-ouia-component-type="PF4/Button"
                                            data-ouia-safe={true}
                                            disabled={false}
                                            onClick={[Function]}
                                            tabIndex={null}
                                            type="button"
                                          >
                                            Download PDF
                                          </button>
                                        </Component>
                                      </DownloadButton>
                                    </DownloadExecutive>
                                  </div>
                                </Component>
                              </article>
                            </Component>
                          </div>
                        </Component>
                        <Component
                          span={3}
                        >
                          <div
                            className="pf-l-grid__item pf-m-3-col"
                          >
                            <Component>
                              <article
                                className="pf-c-card"
                              >
                                <Component>
                                  <div
                                    className="pf-c-card__title"
                                  >
                                    <FileAltIcon
                                      color="var(--pf-global--link--Color)"
                                      noVerticalAlign={false}
                                      size="lg"
                                    >
                                      <svg
                                        aria-hidden={true}
                                        aria-labelledby={null}
                                        fill="var(--pf-global--link--Color)"
                                        height="2em"
                                        role="img"
                                        style={
                                          Object {
                                            "verticalAlign": "-0.25em",
                                          }
                                        }
                                        viewBox="0 0 384 512"
                                        width="2em"
                                      >
                                        <path
                                          d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm64 236c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-64c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-72v8c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12zm96-114.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
                                          transform=""
                                        />
                                      </svg>
                                    </FileAltIcon>
                                    <span
                                      className="pf-u-ml-sm"
                                      style={
                                        Object {
                                          "verticalAlign": "0.3rem",
                                        }
                                      }
                                    >
                                      Report by CVEs
                                    </span>
                                  </div>
                                </Component>
                                <Component>
                                  <div
                                    className="pf-c-card__body"
                                  >
                                    A customizable PDF report of vulnerabilities identified by Red Hat that may impact your RHEL servers.
                                  </div>
                                </Component>
                                <Component>
                                  <div
                                    className="pf-c-card__footer"
                                  >
                                    <a
                                      onClick={[Function]}
                                    >
                                      Create report
                                    </a>
                                  </div>
                                </Component>
                              </article>
                            </Component>
                          </div>
                        </Component>
                      </div>
                    </Component>
                  </section>
                </Main>
              </Connect(Main)>
              <ReportConfigModal
                isOpen={false}
                onClose={[Function]}
              >
                <Modal
                  actions={
                    Array [
                      <DownloadCVEsReport
                        buttonProps={
                          Object {
                            "isDisabled": false,
                            "style": Object {
                              "marginRight": "0.5em",
                            },
                          }
                        }
                        filters={Object {}}
                        isReportDynamic={true}
                        label={
                          Object {
                            "defaultMessage": "Export report",
                            "id": "configModalExportReport",
                          }
                        }
                        params={
                          Object {
                            "business_risk_id": "",
                            "cvss_from": 0,
                            "cvss_to": 10,
                            "impact": "",
                            "public_from": undefined,
                            "public_to": undefined,
                            "publish_date": "all",
                            "rule_presence": undefined,
                            "status_id": "",
                          }
                        }
                        reportData={
                          Object {
                            "header": Array [
                              "publish_date",
                              "impact",
                              "cvss_filter",
                              "show_all",
                              "business_risk_id",
                              "status_id",
                              "security_rule",
                            ],
                            "reportTitle": "Insights Vulnerability CVE Report",
                            "userNotes": "",
                          }
                        }
                        type="cves"
                      />,
                      <Unknown
                        onClick={[Function]}
                        variant="secondary"
                      >
                        Cancel
                      </Unknown>,
                    ]
                  }
                  appendTo={
                    <body>
                      <div />
                    </body>
                  }
                  aria-describedby=""
                  aria-label=""
                  aria-labelledby=""
                  className=""
                  hasNoBodyWrapper={false}
                  isOpen={false}
                  onClose={[Function]}
                  showClose={true}
                  title="Report by CVEs"
                  variant="small"
                >
                  <Portal
                    containerInfo={<div />}
                  >
                    <Component
                      actions={
                        Array [
                          <DownloadCVEsReport
                            buttonProps={
                              Object {
                                "isDisabled": false,
                                "style": Object {
                                  "marginRight": "0.5em",
                                },
                              }
                            }
                            filters={Object {}}
                            isReportDynamic={true}
                            label={
                              Object {
                                "defaultMessage": "Export report",
                                "id": "configModalExportReport",
                              }
                            }
                            params={
                              Object {
                                "business_risk_id": "",
                                "cvss_from": 0,
                                "cvss_to": 10,
                                "impact": "",
                                "public_from": undefined,
                                "public_to": undefined,
                                "publish_date": "all",
                                "rule_presence": undefined,
                                "status_id": "",
                              }
                            }
                            reportData={
                              Object {
                                "header": Array [
                                  "publish_date",
                                  "impact",
                                  "cvss_filter",
                                  "show_all",
                                  "business_risk_id",
                                  "status_id",
                                  "security_rule",
                                ],
                                "reportTitle": "Insights Vulnerability CVE Report",
                                "userNotes": "",
                              }
                            }
                            type="cves"
                          />,
                          <Unknown
                            onClick={[Function]}
                            variant="secondary"
                          >
                            Cancel
                          </Unknown>,
                        ]
                      }
                      aria-describedby=""
                      aria-label=""
                      aria-labelledby=""
                      boxId="pf-modal-part-0"
                      className=""
                      descriptorId="pf-modal-part-2"
                      hasNoBodyWrapper={false}
                      isOpen={false}
                      labelId="pf-modal-part-1"
                      onClose={[Function]}
                      showClose={true}
                      title="Report by CVEs"
                      variant="small"
                    />
                  </Portal>
                </Modal>
              </ReportConfigModal>
            </main>
          </div>
        </Page>
      </ReportsPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
